### PR TITLE
MS-134: Clients implement redux

### PR DESCRIPTION
**Pull request by valeferaudo**

### CODE REVIEW

File: src/Components/Clients/Form/index.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+import { useSelector, useDispatch } from 'react-redux';
 import styles from './form.module.css';
 import Spinner from '../../Shared/Spinner';
 import Input from '../../Shared/Input';
 import Button from '../../Shared/Button';
+import { getOneClients } from '../../../redux/Clients/thunks';
 
 function ClientsForm({ id, handleSubmit, handleShowModal }) {
-  const [isLoadingForm, setLoadingForm] = useState(false);
+  // const client = useSelector((store) => store.clients.client);
```
Comment (sofpag) [inline src/Components/Clients/Form/index.js:10]: Maybe you could delete those coments

File: src/Components/Clients/index.js [hunk 2]
```diff
@@ -1,154 +1,74 @@
...
+  }, [dispatch]);
 
   const handleClickDelete = (id) => {
-    cleanMessage();
-    setShowModal(true);
+    dispatch(setShowModal(true));
     setIdActive(id);
-    setModalType('delete');
+    dispatch(setModalType('delete'));
+    console.log(id);
```
Comment (sofpag) [inline src/Components/Clients/index.js:30]: and delete this console.log too

File: src/redux/Clients/actions.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { MODAL_TYPE, SHOW_MODAL } from '../../constants/actionTypes';
+import { SHOW_MESSAGE } from '../../constants/actionTypes';
```
Comment (sofpag) [inline src/redux/Clients/actions.js:2]: you could do only one line of code here

File: src/redux/Clients/reducer.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  MODAL_TYPE
+} from '../../constants/actionTypes';
+const initialState = {
+  isLoading: false,
+  isLoadingForm: false,
+  list: [],
+  error: false,
+  messageType: '',
+  messageText: '',
+  client: null,
```
Comment (sofpag) [inline src/redux/Clients/reducer.js:28]: to me put  "null" cause me trouble, so I choose to put  an objet (" { } ") as I saw do Julián

File: src/redux/Clients/reducer.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        ...state,
+        isLoading: false,
+        error: true,
+        messageType: 'error',
+        messageText: 'Cannot delete Clients'
+      };
+    case UPDATE_CLIENTS_FETCHING:
+      return {
+        ...state
+        // isLoading: true
```
Comment (sofpag) [inline src/redux/Clients/reducer.js:98]: you left many comments in this file

File: src/redux/Clients/thunks.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
+import {
+  GET_CLIENTS_FETCHING,
+  GET_CLIENTS_FULFILLED,
+  GET_CLIENTS_REJECTED,
+  ADD_CLIENTS_FETCHING,
+  ADD_CLIENTS_FULFILLED,
+  ADD_CLIENTS_REJECTED,
+  UPDATE_CLIENTS_FETCHING,
+  UPDATE_CLIENTS_FULFILLED,
```
Comment (sofpag) [inline src/redux/Clients/thunks.js:9]: this break the consistency, in the other files you import in this order: get, add, DELETE ...

File: src/redux/Clients/thunks.js [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+});
+
+const getClientsRejected = () => ({
+  type: GET_CLIENTS_REJECTED
+});
+
+export const getClients = () => {
+  return (dispatch) => {
+    dispatch(getClientsFetching());
+    fetch(BASE_URL)
```
Comment (sofpag) [inline src/redux/Clients/thunks.js:37]: as I said on Slack, i believe we could put more about the status error when we do the "fetch"

**Review by sofpag — CHANGES_REQUESTED** [review]

**Comment by sofpag** [comment]
@valeferaudo The ADD didn´t work on herolu nor on local. 
This is in local 
imagen
imagen

Also GET_ONE is not working on local (but it is working on heroku)
imagen

So, I could not try the EDIT on local, but on heroku works

DELETE look fine

**Review by BareaMartin — APPROVED** [review]
Tested all on heroku deployment, looks good to me.

**Review by sofpag — APPROVED** [review]

**Pull request merged**